meta/store.go

```diff
@@ -735,12 +734,20 @@ func (s *Store) serveExecListener() {
 		// Accept next TCP connection.
 		var err error
 		conn, err := s.ExecListener.Accept()
-		if err != nil {
-			if strings.Contains(err.Error(), "connection closed") {
+		if opErr, ok := err.(*net.OpError); ok && opErr.Temporary() {
```

Comment:
OK, I see where you are going.

This still seems a little bit awkward to me. Let me explain.

If the error returned from `Accept` is temporary, then keep looping. However, if the error is not temporary, then return. The way other code signals that the function should return is simply by closing the listener. That will result in a non-temporary error, so polling `closing` isn't really necessary.

Perhaps this wouldn't work with the `ExecListener` for some reason? (I have not looked at that code yet).